docs/_data/xpath_funs.yml

```diff
@@ -172,3 +172,18 @@ langs:
             outcome: "Matches calls to any method with 2 `int` parameters (!= argument)"
           - code: '//ConstructorCall[pmd-java:matchesSig("java.util.ArrayList#new(int)")]'
             outcome: "Matches constructors calls of ArrayList with a single int parameter"
+
+  - name: "Kotlin"
+    ns: "pmd-kotlin"
+    funs:
+      - name: hasChildren
+        returnType: "xs:boolean"
+        shortDescription: "Tests if a node has children"
+        description: "Returns true if the Node has children. This is for test purposes only."
```

Comment:
I don't remember the context for this, but it should be removed. Why would one use `pmd-kotlin:hasChildren(3)` over pure XPath `count(*) = 3`

Reply:
True, can be removed and pure XPath can be used. I used this to check how and if XPath functions for Kotlin are working in pmd-designer and XPath Kotlin rules. I just checked again and it works in pmd-designer  (compat-7.0 branch). Earlier I had troubles with "Namespace prefix: 'pmd-kotlin' has not been declared.". 

Maybe we can think of a better/more useful XPath function to use.

Reply:
<img width="896" alt="image" src="https://user-images.githubusercontent.com/8797018/156189360-1801615f-5704-43b2-9cb0-32aef0814d9a.png">

But I noticed that only `foo.ba` is selected, not `foo.bar`: something for pmd-designer or pmd-kotlin module?

Reply:
> Maybe we can think of a better/more useful XPath function to use. For instance to combine text of the children to replace the deprecated @joinTokenText XPath attribute?

The attribute is deprecated *because* it joins the token text and is therefore very inefficient. The deprecation is a deterrent against using that in rules. PMD rules shouldn't rely on the text of nodes but rather on the structure of the AST.

> But I noticed that only foo.ba is selected, not foo.bar: something for pmd-designer or pmd-kotlin module?

This is a bug I fixed recently on the compat-7.0 branch https://github.com/pmd/pmd-designer/commit/775d6537a0010b257a7a4d99bed7d3789f6cf9ba

Reply:
I've remove the XPath function now